src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -153,7 +153,7 @@ private[impl] class InstanceTrackerActor(
 
         updateEffect match {
           case InstanceUpdateEffect.Update(instance, _, _) =>
-            repository.store(instance).onComplete {
+            repository.store(state.Instance.fromCoreInstance(instance)).onComplete {
```

Comment:
I this could be the interface I wrote in one of the previous comments... It would accept core instances and give you core instances and all the transformation will happen behind the curtains... As a consumer of repository I don't really care about how the concrete serialization format looks like

Reply:
I think we were there two years ago. Back then I thought it would be nice to have storage models the same way we have API models. I do not think that the repositories should return the internal models but the storage models. In any case, I'm happy to create a wrapper repository on the `InstanceRepository` and `GroupRepository` that converts the storage model to the internal model. IMHO that is some unnecessary overhead until we really need more joins like these. I mean, the `InstanceLoderImpl` is doing exactly what you want for loads at least. Would you be fine if that becomes a layer between our core logic and the repositories?